src/widgets/reflow.rs

```diff
@@ -443,4 +450,23 @@ mod test {
         let (word_wrapper_space, _) = run_composer(Composer::WordWrapper, &text_space, width);
         assert_eq!(word_wrapper_space, vec!["AAAAAAAAAAAAAAA AAAA", "AAA",]);
     }
+
+    #[test]
+    fn line_composer_word_wrapper_preserve_indentation() {
+        let width = 20;
+        let text = "AAAAAAAAAAAAAAAAAAAA    AAA";
+        let (word_wrapper, _) = run_composer(Composer::WordWrapperNoTrim, text, width);
+        assert_eq!(word_wrapper, vec!["AAAAAAAAAAAAAAAAAAAA", "   AAA",]);
+    }
+
+    #[test]
+    fn line_composer_word_wrapper_preserve_indentation_with_wrap() {
+        let width = 10;
+        let text = "AAA AAA AAAAA AA AAAAAA\n B\n  C\n   D";
```

Comment:
Shouldn't we have a case of a line leading with white-spaces but whose length is greater than the width ?

Reply:
A very good catch! I'll add one! Sorry for missing that the first time around.